export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+    raise TimeOptionError('ts_current - ts_end < min offset: %s - %s < %s' %
+        (ts_current, ts_end, options.ts_offset))
+  return ts_end
+
+
+def default_start_time(options, ts_previous):
+  """Calculates a default start timestamp.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.step, options.update, options.ts_end.
+    ts_previous: int, timestamp in seconds since epoch of last
+        successful export. On first run, this value should be zero.
+  Returns:
+    int, timestamp in seconds since the epoch.
+  """
+  # Only when both options.update and ts_previous are set, skip assignment.
+  if not (options.update and ts_previous):
+    # If either update or ts_previous were not set, then choose a default.
+    ts_previous = options.ts_end - options.length
+
+  return align_timestamp(ts_previous, options.step)
+
+
+def assert_start_and_end_times(options):
+  """Performs a sanity check on start and end timestamps.
+
+  This method asserts that both ts_end is less than ts_start and that the
+  difference between them is greater than options.length.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.ts_start, options.ts_end.
+  Raises:
+    TimeOptionError, if a start & end time constraint is violated.
+  """
+  # Always check if basic constratins are respected.
+  if options.ts_end < options.ts_start:
+    raise TimeOptionError('Start time cannot be later than end time.')
+
+  if options.ts_end - options.ts_start < options.length:
+    msg = ('Start and end times difference is less than length: '+
+           '%s - %s = %s < %s' % (options.ts_end, options.ts_start,
+           (options.ts_end - options.ts_start), options.length))
+    raise TimeOptionError(msg)
+
+  logging.debug('Exporting: %s to %s', time.ctime(options.ts_start),
+                time.ctime(options.ts_end))
+
+
+def default_output_name(ts_start, ts_end, output_dir):
+  """Creates a default output filename based on time range and output dir.
+
+  Filenames are formatted with time stamps as:
+      <output_dir>/resource-utilization/YYYY/MM/DD/<HOSTNAME>/
+          metrics-YYYYMMDDTHHMMSS-to-YYYYMMDDTHHMMSS.json
+
+  Args:
+    ts_start: int, starting timestamp of export in seconds since the epoch.
+    ts_end: int, ending timestamp of export in seconds since the epoch.
+    output_dir: str, base path of directory for output.
+  Returns:
+    str, absolute path of generated output file name.
+  """
+  filename = 'metrics-%s-to-%s.json' % (
+      time.strftime('%Y%m%dT%H%M%S', time.localtime(ts_start)),
+      time.strftime('%Y%m%dT%H%M%S', time.localtime(ts_end)))
+  date_path = time.strftime('%Y/%m/%d', time.localtime(ts_start))
+  full_path = os.path.join(
+      output_dir, 'resource-utilization', date_path, HOSTNAME)
+  if not os.path.exists(full_path):
+    os.makedirs(full_path)
```

Comment:
Modifying the filesystem in this function is a bit unexpected. Suggest mentioning this in the docstring.